recipes/libgcrypt/all/conanfile.py

```diff
@@ -0,0 +1,84 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.33.0"
+
+class LibgcryptConan(ConanFile):
+    name = "libgcrypt"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.gnupg.org/download/index.html#libgcrypt"
+    description = "Libgcrypt is a general purpose cryptographic library originally based on code from GnuPG"
+    topics = ("conan", "libgcrypt", "gcrypt", "gnupg", "gpg", "crypto", "cryptography")
+    license = "LGPL-2.1-or-later"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    generators = "cmake"
+    requires = 'libgpg-error/1.36'
+    exports_sources = "CMakeLists.txt", "config.h.in", "gcrypt.h.in", "mod-source-info.h", "patches/**"
+
+    _cmake = None
```

Comment:
libgcrypt provides a autotools build suite.
Since you are already excluding all os'es except Linux, can't you just use that?

Reply:
@madebr, good question, and I have several thoughts about this. The first is that I borrowed recipe from bincrafters team and slightly modified their recipe to meet the requirements of CCI. Recipe of bincrafters initially uses cmake based build system, I haven't risked to modify their code so serious. The second is that I personally hate autotools and I believe it should disappear in the future at all, I support cmake-based build systems, and if there is a chance to advertise cmake I will

Reply:
AFAIK we were trying to port recipe to MSVC and MinGW, and with autotools, were unable to do so (don't remember why exactly), but with CMake build we were able to achieve a significant progress.

Reply:
Your "hate" is well-founded and shared with a lot of people here, but we don't like to maintain a build system ourselves.
Since we will not support other os'es then Linux initially, I think the recipe can just use autotools.

Reply:
And it can be done in another PR, can't it?

Reply:
You're only adding support for Linux, so there's no actual need for crossbuild support with cmake.

Reply:
@madebr, ok, thank you, maybe we should reach out to @SSE4. @SSE4, as far as I have managed to understand you've been a member of bincrafters team, haven't you? What do you advise? I admit to steal your team's recipe. Should I swap cmake for autotools for this recipe?

Reply:
I would leave CMake for now.
yes, I was a member of bincrafters team.
for my knowledge, we were able to compile gpg-error for MinGW and MSVC, but with CMake.
why current recipe only supports linux, I don't remember, maybe it was another branch.
let's proceed with CMake for now. and then someone (maybe myself) will add MSVC/MinGW support in another PR.

Reply:
Current recipe supports only Linux because I have restricted. I didn't manage to support Windows for libgpg-error because for Windows it requires to pull several packages more in, like autotools_aux or something like that, and these packages exist only in bincrafters repositories, and they are suspicious for being merged in CCI, I doubt they will be accepted easily. To be honest, also I do not like to pull 1 million packages to support just one library libgcrypt for all the operating systems, for me Linux will do at now

Reply:
yes, that's totally fine. you don't have to support all the possible configurations and platforms in the first iteration. this could be safely added later, in another PRs.